distribution/src/bin/elasticsearch-autoconfig-security

```diff
@@ -0,0 +1,7 @@
+#!/bin/bash
```

Comment:
If we have a way to expose this without a CLI tool of its own so that we can call it from postinst, we should do it - I wasn't aware of such a way. If this needs to stay, can we disallow users from calling it directly? Should we ?

Reply:
I don't think we have anything like this. You could add a mandatory env var or command line option to discourage casual use e.g. `--i-know-what-im-doing` or `--this-is-risky` 😁

We should at least include a comment in the script itself and in the `--help` text that users should not run the tool themselves.

Actually, I suppose we could put the tool somewhere other than `./bin`, such as `lib/tools/security-cli/`. It doesn't stop a user calling it, but by putting it somewhere outside `bin`, we suggest that it's not for normal use. @mark-vieira what do you think?

Reply:
> somewhere other than ./bin, such as lib/tools/security-cli/. It doesn't stop a user calling it, but by putting it somewhere outside bin, we suggest that it's not for normal use.

I like this idea!

Reply:
I mean, all this shell script does is call `java` with a certain classpath. We could just embed this in `postinstall` itself. That said, there still exists the question of whether we _should_ explicitly expose this. @colings86 and I specifically spoke about this today. There is a use case where a user upgrading explicitly disables security, and then leverages this tool to enable it later. I think it's probably worth while having this available to users. That said, we'd want to document it as we would any other CLI tool.

Reply:
> I mean, all this shell script does is call java with a certain classpath. We could just embed this in postinstall itself. 

Yes @albertzaharovits made a similar comment when we discussed and we had already agreed to do this. 

> That said, there still exists the question of whether we should explicitly expose this. @colings86 and I specifically spoke about this today. There is a use case where a user upgrading explicitly disables security, and then leverages this tool to enable it later. I think it's probably worth while having this available to users. That said, we'd want to document it as we would any other CLI tool.


Yes, this is a relevant discussion. We had this in the es-security team and we agreed to look into exposing this as a standalone tool too ( with some different bahavior ) and maybe backport this to 7.x too.